File: backend-play/app/com/github/brane08/pagila/store/entities/Staff.java
package com.github.brane08.pagila.store.entities;

import com.github.brane08.pagila.seedworks.entities.BaseModel;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "staff")
public class Staff extends BaseModel {

	@Id
	@Column(name = "staff_id")
	public Integer staffId;

	@Column(name = "first_name")
	public String firstName;

	@Column(name = "last_name")
	public String lastName;

	@Column(name = "email")
	public String email;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "store_id")
	public Store store;

	@Column(name = "active")
	public boolean active;

	@Column(name = "username")
	public String username;

	@Column(name = "password")
	public String password;

	@Column(name = "picture")
	public byte[] picture;

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Staff that = (Staff) o;
		return staffId.equals(that.staffId);
	}

	@Override
	public int hashCode() {
		return Objects.hash(staffId);
	}
}

File: backend-micronaut/src/main/kotlin/com/github/brane08/pagila/models/Staff.kt
package com.github.brane08.pagila.models

import javax.persistence.*

@Entity
@Table(name = "staff")
class Staff : BaseModel() {
	@Id
	@Column(name = "staff_id")
	var staffId: Int? = null

	@Column(name = "first_name")
	var firstName: String? = null

	@Column(name = "last_name")
	var lastName: String? = null

	@Column(name = "email")
	var email: String? = null

	@ManyToOne(optional = false)
	@JoinColumn(name = "store_id")
	var store: Store? = null

	@Column(name = "active")
	var active = false

	@Column(name = "username")
	var username: String? = null

	@Column(name = "password")
	var password: String? = null

	@Column(name = "picture")
	var picture: ByteArray? = null
}